docs/src/app/(public)/(content)/react/handbook/composition/page.mdx

```diff
@@ -0,0 +1,82 @@
+# Composition
+
+<Subtitle>A guide to composing Base UI components with your own React components.</Subtitle>
+<Meta
+  name="description"
+  content="A guide to composing Base UI components with your own React components."
+/>
+
+### Composing custom React components
+
+Use the `render` prop to compose a Base UI part with your own React components.
+
+For example, most triggers render a `<button>` by default. The code snippet below shows how to use a custom button instead.
+
+```tsx title="index.tsx"
+<Menu.Trigger render={<MyButton size="md">Open menu</MyButton>} />
+```
+
+The custom component must forward the `ref`, and spread all the received props on its underlying DOM node.
+
+### Composing multiple components
+
+In situations where you need to compose multiple Base UI components with custom React components, `render` props can be nested as deeply as necessary. Working with Tooltip is a common example.
+
+```tsx title="index.tsx"
+<Dialog.Root>
+  <Tooltip.Root>
+    <Tooltip.Trigger
+    {/* prettier-ignore */}
+      render={
+        <Dialog.Trigger
+          render={
+            <MyButton size="md">Open dialog</MyButton>
+          }
+        />
+      }
+    />
+    <Tooltip.Portal>...</Tooltip.Portal>
+  </Tooltip.Root>
+  <Dialog.Portal>...</Dialog.Portal>
+</Dialog.Root>
+```
+
+### Changing the default rendered element
+
+You can also use the `render` prop to override the rendered element of the component.
+
+For example, `Menu.Item` renders a `<div>` by default. The code snippet below shows how to render it as an `<a>` element so that it works like a link.
+
+```tsx title="index.tsx"
+import * as React from 'react';
+import { Menu } from '@base-ui-components/react/menu';
+
+export default () => (
+  <Menu.Root>
+    <Menu.Trigger>Song</Menu.Trigger>
+    <Menu.Portal>
+      <Menu.Positioner>
+        <Menu.Popup>
+          <Menu.Item render={<a href="base-ui.com" />}>Add to Library</Menu.Item>
+        </Menu.Popup>
+      </Menu.Positioner>
+    </Menu.Portal>
+  </Menu.Root>
+);
+```
+
+Each Base UI component renders the most appropriate element by default, and in most cases, rendering a different element is recommended only on a case-by-case basis.
+
+### Render function
+
+If you are working in an extremely performance-sensitive application, you might want to use the function `render` prop instead of the element `render` prop.
```

Comment:
This doesn't compute for me

```suggestion
If you are working in an extremely performance-sensitive application, you might want to pass a function to the `render` prop instead of a React element.
```

Reply:
https://github.com/mui/base-ui/pull/1141#discussion_r1888656196 (it's collapsed above, buried between other "outdated" comments)